laikaIO/src/main/scala/pink/cozydev/protosearch/analysis/PlaintextRenderer.scala

```diff
@@ -56,15 +53,17 @@ object PlaintextRenderer extends ((TextFormatter, Element) => String) {
   }
 }
 
-case object Plaintext extends RenderFormat[TextFormatter] {
+case object Plaintext extends RenderFormat[Formatter] {
   val fileSuffix = "txt"
 
-  val defaultRenderer: (TextFormatter, Element) => String = PlaintextRenderer
+  val defaultRenderer: (Formatter, Element) => String =
+    PlaintextRenderer
 
-  val formatterFactory: RenderContext[TextFormatter] => TextFormatter = PlaintextFormatter
+  val formatterFactory: Formatter.Context[Formatter] => Formatter =
```

Comment:
Very minor thing: to reduce a bit if boilerplate you could also simply inline this like here:
https://github.com/typelevel/Laika/blob/main/core/shared/src/main/scala/laika/format/AST.scala#L41